Add proposal autocomplete in editor

mllocs opened the pull request:
#### :tophat: What? Why?
Adds ability to mention a Proposal from the richtext editor, similar to the user mentions or hashtags. The Autocomplete pops up when the `/` key is used. 

#### :pushpin: Related Issues
- Fixes https://github.com/decidim/decidim/issues/14151

#### Testing

- User can mention a proposal by using the `/` autocomplete extension in the rich text editor. https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/5/manage/meetings/21/edit

- A proposal mention renders with the name of the proposal, linking the proposal page, preceded with the avatar of the first author: https://decidim-lot2.populate.tools/processes/branch-highway/f/5/meetings/21

### :camera: Screenshots

![CleanShot 2025-03-05 at 14 53 07@2x](https://github.com/user-attachments/assets/05d915f6-f36a-4f09-a9c4-db513755e0a9)

![CleanShot 2025-03-05 at 14 53 44@2x](https://github.com/user-attachments/assets/ae9a6c86-46da-4fb1-b8a1-0e3bc01f4998)


:hearts: Thank you!

NilHomedes reviewed (changes requested):
Hi @mllocs!
Thanks for the work, looks awesome :sparkles: 

Just a small thing:
- [x] When I write something before the proposal mention it automatically creates a break line. Could we avoid this line break?
![imatge](https://github.com/user-attachments/assets/5c0d4297-5af2-4845-a337-6af12e84b060)
- [x] In the previous image, why is retrieving this avatar image? It should be the default avatar. This is the proposal: https://decidim-lot2.populate.tools/processes/branch-highway/f/4/proposals/10

mllocs commented:
> Hi @mllocs! Thanks for the work, looks awesome ✨
> 
> Just a small thing:
> 
> ...

@NilHomedes done!

NilHomedes reviewed (changes requested):
Hi @mllocs thanks for the updates
Here is my last review:

- [x] Change the command from `$` to `/` 
- [ ] For consistency, in the front-end, the proposals linked should be bold and underlined. See an example: 
![imatge](https://github.com/user-attachments/assets/ff361086-e944-4c19-b034-7ed763c43a5a)

NilHomedes reviewed (changes requested):
Hi @mllocs thanks for the updates
Here is my last review:

- [x] Change the command from `$` to `/` 
- [ ] For consistency, in the front-end, the proposals linked should be bold and underlined. See an example: 
![imatge](https://github.com/user-attachments/assets/ff361086-e944-4c19-b034-7ed763c43a5a)

entantoencuanto reviewed (changes requested):
Great work! I would add an integration test mentioning a proposal from a meeting body for example.
I have found a small issue: when the proposal is selected by clicking on its title the last character remains.

[Screencast from 14-03-25 19:08:03.webm](https://github.com/user-attachments/assets/f87ae1bb-ea54-4678-86f1-5ffbb32962ce)

mllocs commented:
> Great work! I would add an integration test mentioning a proposal from a meeting body for example. I have found a small issue: when the proposal is selected by clicking on its title the last character remains.
> ...

Looks like this is a common issue to all the autocompleters (mentions, hashtags, and now resource mentions). There is already a "fix" in place but it's not working. Maybe you can guide us regarding this @ahukkanen? 🙏

furilo commented:
@NilHomedes  good to go?

NilHomedes reviewed (changes requested):
Hi @mllocs! Thanks for the changes, now working as expected.

- [ ] I noticed that in the meeting's Agenda, proposal are not rendering as links, see:
![image](https://github.com/user-attachments/assets/1f214319-930d-4c39-a470-f82a4cf154d8)

mllocs commented:
> Hi @mllocs! Thanks for the changes, now working as expected.
> 
> * [ ]  I noticed that in the meeting's Agenda, proposal are not rendering as links, see:
> ...

Fixed @NilHomedes. After my fix, I had to recreate the mention to the proposal, and I just fixed this one: https://decidim-lot2.populate.tools/processes/branch-highway/f/5/meetings/21

Saying that in case you did other tests!

NilHomedes reviewed (approved):
Awesome, workinbg as expected! Thanks @mllocs! :sparkles:

mllocs commented:
@alecslupu ready for you :)

alecslupu reviewed (changes requested):
I just check this and i have some suggestions : 

1. in the PR description you wrote that "$" is triggering the autocoplete, but the key is "/" 
2. As a result of first point, i would like to see a help message telling me "you can press .. to autocomplete"



1. I went to a meeting body and tried to write something like "This should work with meetings / proposals / debates " which actually started to prompt for autocomplete all the following text ( like 1 request of each one of the chars after "meeting /" 
2. I went to organization settings, and tried the same text and tried to autocomplete 
3. When i have tried to press enter, my text vanished ( like the prompt did not fined anything, and cleared everything )
4. In some edge case, when i pressed enter, i could not get to the next page

cc @NilHomedes , @furilo

mllocs commented:
> I went to organization settings, and tried the same text and the system tried to do the autocomplete ( aka fired the autocomplete wizard )

@alecslupu can you explain this one? Where exactly did you try this?

alecslupu commented:
> @alecslupu can you explain this one? Where exactly did you try this?

I went to the admin , Settings / Configuration in TOS field.

mllocs commented:
@alecslupu  I think that all you mentioned is fixed now.

NilHomedes commented:
Hi @mllocs! 
It's okay to restrict this feature to specific parts. However, we need it in other parts above of meetings. I share with you the complete list:

- Proposals 
- Debates
- Meetings
- About (participatory space)

This feature will only be able for admins, so participants couldn't trigger the proposal autocomplete from the front-end.

mllocs commented:
> Hi @mllocs! It's okay to restrict this feature to specific parts. However, we need it in other parts above of meetings. I share with you the complete list:
> 
> * Proposals
> * Debates
> * Meetings
> * About (participatory space)
> 
> This feature will only be able for admins, so participants couldn't trigger the proposal autocomplete from the front-end.

Done!

NilHomedes commented:
Sorry I forget Budgets, could we add them too? 
@mllocs With that, the feature looks good to me.

alecslupu commented on decidim-participatory_processes/app/views/decidim/participatory_processes/admin/participatory_processes/_form.html.erb:
```diff
@@ -32,11 +32,11 @@
 
       <div class="row column">
-        <%= form.translated :editor, :short_description, aria: { label: :short_description } %>
+        <%= form.translated :editor, :short_description, resource_mentionable: true, aria: { label: :short_description } %>
```
This is not working as it should. I can mention the resource, but in  the frontend the mention si not linked. 
![image](https://github.com/user-attachments/assets/5d486571-3387-428c-835c-2403aac5165b)

mllocs replied:
@alecslupu, fixed now. I thought most of the rich text fields had the processors in place but I guess only those that already support mentions or hashtags do. You'll have to do the mentions again because the once you created were not processed correctly.

mllocs replied:
oh wait, looking at the failing tests, I'll have to do a fix.

alecslupu reviewed (changes requested):
Good Job! All the previously mentioned issues are fixed, but I have 2 additional observations. 

Also, even though it might be out of the scope of this implementation, i would like to see this applied to assemblies and conferences just to make sure it is consistent.

alecslupu reviewed (approved):
:+1:
